dashboard/app/controllers/api/v1/sections_students_controller.rb

```diff
@@ -61,6 +61,22 @@ def bulk_add
     end
   end
 
+  # Remove a student from the section
+  # POST /sections/:section_id/students/:id/remove
+  def remove
+    student = @section.students.find_by_id(params[:id])
+    follower = Follower.where(section: @section.id, student_user_id: student.id).first
```

Comment:
It seems a little odd that we need the Follower, vs. just something like `@section.remove_student(@student)`, but according to the comments above [Section.remove_student](https://github.com/code-dot-org/code-dot-org/blob/2565171417c45d00b273834cf2bcb1ab5d62af15/dashboard/app/models/sections/section.rb#L186-L189), it's for authorization, which isn't being done here. 

We either should authorize it, or change the API to not require it. 

Note we've already authorized the section in `load_and_authorize_resource`. 
According to `ability.rb`, teachers can manage (which means they can perform any action on) [all followers](https://github.com/code-dot-org/code-dot-org/blob/ceb38fb79dedd3cd661e250de73a8a8f14875071/dashboard/app/models/ability.rb#L80) and [sections they own](https://github.com/code-dot-org/code-dot-org/blob/ceb38fb79dedd3cd661e250de73a8a8f14875071/dashboard/app/models/ability.rb#L75). In addition all users can [create and destroy followers in which they are the student](https://github.com/code-dot-org/code-dot-org/blob/ceb38fb79dedd3cd661e250de73a8a8f14875071/dashboard/app/models/ability.rb#L63-L64).

So, since we've already authorized the section, we know this is the teacher who owns the section. Follower is even more permissive, allowing any teacher to manage it, so I don't see any point given our current auth definition of authorizing the follower.

Reply:
I think what you're saying makes sense, but I think I'll make this a followup/tech debt item since it looks like this function is used with the follower in other places in code and I don't want to mess with that right now.